src/Elewant/Trading/DomainModel.md

```diff
@@ -0,0 +1,84 @@
+Welcome to the Trading domain
+-----------------------------
+
+
+### Offers
+
+Here, we start with an `Offer`. An `Offer` is provided by a `seller`, who is represented here by nothing
+more than a `SellerId`.
+
+When an offer is `placed`, it is in reference of an `ElePHPant`. An ElePHPant consists of an `ElePHPantId`
+and a `Breed`. Since we do not want to tie the Trading and Herding contexts together, it is not required for
```

Comment:
elePHPants are already unique by their `ElePHPantId`; why is the `Breed` also necessary?

Reply:
Hmm.. good point. When placing an offer (or proposing a trade for that matter), if the ElePHPant is chosen from a Herd in the user-interface, we already know the Id, and so we _could_ go back to that data.
But that would tie those two datasets together, and also force a Seller to always have an ElePHPant in a herd before an offer can be created.

The proposal works like this: If you select an ElePHPant from your herd, the elePHPant data is "copied" to a new ElePHPant (consisting of the same data) in the Trading domain. But: you can also just say "I want to offer an ElePHPant of breed X", and then we create a new ElePHPant with a new GUID of that breed that does not exist in any herd.

One of the main drivers of the Elewant project is to learn about loose coupling between bounded contexts, so all those decisions might be bad and wrong :D, but finding that out is valuable in itsself ;)

Reply:
This issue is one example where the origin of an elephpant available for sale/trade might not be an elephpant in a Herd persé:

https://github.com/f500/elewant/issues/176